test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/io/CharArrayWriterTest.scala

```diff
@@ -0,0 +1,227 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+ package org.scalajs.testsuite.javalib.io
+
+import java.io._
+ 
+import org.junit.Assert._
+import org.junit.{After, Before, Test}
+
+// Adapted from Apache Harmony CharArrayWriterTest.java
+//   https://github.com/apache/harmony/blob/02970cb7227a335edd2c8457ebdde0195a735733/classlib/modules/luni/src/test/api/common/org/apache/harmony/luni/tests/java/io/CharArrayReaderTest.java
+
+class CharArrayWriterTest {
+  private[io] val hw: Array[Char] = Array('H', 'e', 'l', 'l', 'o', 'W', 'o', 'r', 'l', 'd')
+  private[io] var cw: CharArrayWriter = null
+  private[io] var cr: CharArrayReader = null
+
+  /**
+    * @tests java.io.CharArrayWriter#CharArrayWriter()
+    */
+  @Test def test_Constructor(): Unit = {
+    cw = new CharArrayWriter(90)
+    assertEquals("Created incorrect writer", 0, cw.size)
```

Comment:
This should also test calls with negative arguments.

Reply:
added